### PR TITLE
Marshall NSDecimalNumber as Objective-C object wrapper - close #14

### LINKED ISSUE #14: NSDecimalNumber marshalling
Currently `NSDecimalNumber` (which inherits from `NSNumber`) is marshalled as JS number primitive and the precision is lost. We should create a wrapper object instead like we do for `NSString` and `NSMutableString`.

### PR DESCRIPTION
**Pull request by jasssonpet**

### CODE REVIEW

File: tests/TestRunner/app/Marshalling/ObjCTypesTests.js [hunk 1]
```diff
@@ -204,6 +204,14 @@ describe(module.id, function () {
         expect(interop.handleof(buffer)).toBe(data.bytes);
     })
 
+    it("MethodWithNSDecimalNumber", function () {
+        expect(NSDecimalNumber.maximumDecimalNumber() instanceof NSDecimalNumber).toBe(true);
```
Comment (fealebenpae) [inline tests/TestRunner/app/Marshalling/ObjCTypesTests.js]: Technically it'd be the same, but why'd you choose this over `expect(NSDecimalNumber.maximumDecimalNumber().toEqual(jasmine.any(NSDecimalNumber))`?

Comment (jasssonpet) [inline tests/TestRunner/app/Marshalling/ObjCTypesTests.js]: I think that `instanceof` makes the code more clear.

Comment (fealebenpae) [inline tests/TestRunner/app/Marshalling/ObjCTypesTests.js]: You've got me there, but wouldn't the error message would be more specific with the `toEqual` matcher if the expectation failed?

**Comment by fealebenpae** [comment]
:+1:

**Pull request merged**